### DISCUSSION TITLE
Best way to model handheld sensor

### DISCUSSION THREAD

**Discussion by fishingfool** [FraunhoferIOSB/FROST-Server#1935] [Q&A] [answered]
Trying to understand the best way to model a handheld multiparameter sensor (used in water data collection - ph, nitrogen, ..).  We have several HOBO water sensors/monitors at fixed locations that we collect data from - we augment this with data from a handheld monitor that we use to collect additional data at the same locations.  I'd created a separate "Thing" and "Location" for each fixed monitor location - then "Sensor" for each of the types of sensors in the HOBO device (temperature, light), and then created "Datastreams" to tie them together.  Since the handheld multiparameter swaps between the static locations - didn't know if the handheld sensor was best modeled as just multiple sensors tied to fixed "Things" and new "Sensors" and "Datastreams",  or create a separate "Thing"  with new "Sensors" and "Datastreams" but wasn't sure about tying it back to the same but multiple locations? 

If there's a better forum/place to ask/discuss let me know. 
Thanks,

**Comment by hylkevds** [comment, marked as answer]
General modelling questions are good to ask on the [SensorThings GitHub](https://github.com/opengeospatial/sensorthings/discussions)

What to model as the Thing in v1.1 is what you want to attach Datastreams to. In your use case, you probably want to have the time-series for your measurement locations, so those will have to be your things.
You could then model your hand-held device as one or more Sensors, depending if you want to keep all details of the individual sub-sensors of the device separate, or if you can summarise them all in one Sensor entity.

If your device doesn't have and device-specific sensors, like for a battery-level, then you don't need to model the device as a Thing.

**Reply by fishingfool** [reply]
Ok can see that (tho still find a tad confusing) -  like you suggested really want to have the time series attached to location so will handle that way.  Our handheld has multiple sensors so will create one per sensor type.  

In future will post on SensorThings GitHub

**Reply by hylkevds** [reply]
Yes, in v1.1 the data model is not ideal for this use case. 
In version 2.0 of the SensorThings API we are updating the model to support this much better. In V2 one can attach a Datastream to a Feature directly. Images of the draft V2 data model can be found in https://github.com/opengeospatial/sensorthings/issues/167
